pkg/operation/botanist/component/kubeapiserver/kube_apiserver.go

```diff
@@ -445,6 +488,17 @@ func (k *kubeAPIServer) Destroy(ctx context.Context) error {
 		return err
 	}
 
+	if k.values.VPN.HighAvailabilityEnabled {
+		err = kutil.DeleteObjects(ctx, k.client.Client(),
+			k.emptyServiceAccount(),
+			k.emptyRoleHAVPN(),
+			k.emptyRoleBindingHAVPN(),
```

Comment:
You can probably just add them unconditionally to below list since "not found" errors are ignored anyways.